hspec-discover/src/Test/Hspec/Discover/Run.hs

```diff
@@ -62,6 +62,11 @@ mkSpecModule src conf nodes =
   ( "{-# LINE 1 " . shows src . " #-}\n"
   . showString "{-# LANGUAGE NoImplicitPrelude #-}\n"
   . showString "{-# OPTIONS_GHC -fno-warn-warnings-deprecations #-}\n"
+#ifdef MIN_VERSION_GLASGOW_HASKELL
+#if MIN_VERSION_GLASGOW_HASKELL(8,4,0,0)
+  . showString "{-# OPTIONS_GHC -fno-warn-missing-export-lists #-}\n"
+#endif
+#endif
```

Comment:
There are corresponding tests for this code in `RunSpec.hs` that need to be updated.

Reply:
To avoid CPP, I have gone down the export list route.